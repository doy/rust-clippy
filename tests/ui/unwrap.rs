#![warn(clippy::unwrap_used)]

fn unwrap_option() {
    let opt = Some(0);
    let _ = opt.unwrap();
}

fn unwrap_result() {
    let res: Result<u8, ()> = Ok(0);
    let _ = res.unwrap();
}

fn unwrap_get() {
    let v = vec![1, 2, 3];
    let _ = v.get(0).unwrap();
}

fn unwrap_get_mut() {
    let mut v = vec![1, 2, 3];
    let _ = v.get_mut(0).unwrap();
}

#[warn(clippy::get_unwrap)]
fn unwrap_get2() {
    let v = vec![1, 2, 3];
    let _ = v.get(0).unwrap();
}

#[warn(clippy::get_unwrap)]
fn unwrap_get_mut2() {
    let mut v = vec![1, 2, 3];
    let _ = v.get_mut(0).unwrap();
}

#[warn(clippy::get_unwrap)]
#[allow(clippy::unwrap_used)]
fn unwrap_get3() {
    let v = vec![1, 2, 3];
    let _ = v.get(0).unwrap();
}

#[warn(clippy::get_unwrap)]
#[allow(clippy::unwrap_used)]
fn unwrap_get_mut3() {
    let mut v = vec![1, 2, 3];
    let _ = v.get_mut(0).unwrap();
}

fn main() {
    unwrap_option();
    unwrap_result();
    unwrap_get();
    unwrap_get_mut();
    unwrap_get2();
    unwrap_get_mut2();
    unwrap_get3();
    unwrap_get_mut3();
}
