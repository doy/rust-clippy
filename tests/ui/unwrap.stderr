error: used `unwrap()` on `an Option` value
  --> $DIR/unwrap.rs:5:13
   |
LL |     let _ = opt.unwrap();
   |             ^^^^^^^^^^^^
   |
   = note: `-D clippy::unwrap-used` implied by `-D warnings`
   = help: if you don't want to handle the `None` case gracefully, consider using `expect()` to provide a better panic message

error: used `unwrap()` on `a Result` value
  --> $DIR/unwrap.rs:10:13
   |
LL |     let _ = res.unwrap();
   |             ^^^^^^^^^^^^
   |
   = help: if you don't want to handle the `Err` case gracefully, consider using `expect()` to provide a better panic message

error: used `unwrap()` on `an Option` value
  --> $DIR/unwrap.rs:15:13
   |
LL |     let _ = v.get(0).unwrap();
   |             ^^^^^^^^^^^^^^^^^
   |
   = help: if you don't want to handle the `None` case gracefully, consider using `expect()` to provide a better panic message

error: used `unwrap()` on `an Option` value
  --> $DIR/unwrap.rs:20:13
   |
LL |     let _ = v.get_mut(0).unwrap();
   |             ^^^^^^^^^^^^^^^^^^^^^
   |
   = help: if you don't want to handle the `None` case gracefully, consider using `expect()` to provide a better panic message

error: called `.get().unwrap()` on a Vec. Using `[]` is more clear and more concise
  --> $DIR/unwrap.rs:26:13
   |
LL |     let _ = v.get(0).unwrap();
   |             ^^^^^^^^^^^^^^^^^ help: try this: `&v[0]`
   |
   = note: `-D clippy::get-unwrap` implied by `-D warnings`

error: used `unwrap()` on `an Option` value
  --> $DIR/unwrap.rs:26:13
   |
LL |     let _ = v.get(0).unwrap();
   |             ^^^^^^^^^^^^^^^^^
   |
   = help: if you don't want to handle the `None` case gracefully, consider using `expect()` to provide a better panic message

error: called `.get_mut().unwrap()` on a Vec. Using `[]` is more clear and more concise
  --> $DIR/unwrap.rs:32:13
   |
LL |     let _ = v.get_mut(0).unwrap();
   |             ^^^^^^^^^^^^^^^^^^^^^ help: try this: `&mut v[0]`

error: used `unwrap()` on `an Option` value
  --> $DIR/unwrap.rs:32:13
   |
LL |     let _ = v.get_mut(0).unwrap();
   |             ^^^^^^^^^^^^^^^^^^^^^
   |
   = help: if you don't want to handle the `None` case gracefully, consider using `expect()` to provide a better panic message

error: called `.get().unwrap()` on a Vec. Using `[]` is more clear and more concise
  --> $DIR/unwrap.rs:39:13
   |
LL |     let _ = v.get(0).unwrap();
   |             ^^^^^^^^^^^^^^^^^ help: try this: `&v[0]`

error: called `.get_mut().unwrap()` on a Vec. Using `[]` is more clear and more concise
  --> $DIR/unwrap.rs:46:13
   |
LL |     let _ = v.get_mut(0).unwrap();
   |             ^^^^^^^^^^^^^^^^^^^^^ help: try this: `&mut v[0]`

error: aborting due to 10 previous errors

